### PR TITLE
Rework projects table for tenants

### LINKED ISSUE #136: Repurpose project ⇒ tenant
This is needed for #135.

# Story

As a developer of Duffy,
I want that project and admin access are evaluated on the same basis,
so that the code dealing with it doesn’t get too complicated.

# Acceptance Criteria

- [x] ORM classes and associated tables, API endpoints and Python files related to projects are renamed to "tenant(s)", as applicable.
- [x] The new `Tenant` ORM class (and following, the `tenants` table in the database) has a boolean column `is_admin` which would let requests authenticated by their token (yet to come, see #135) manipulate any object, not only those which are related to that tenant.

# Background

I don't know why we started calling these projects in the first place. :wink:

### PR DESCRIPTION
**Pull request by gridhead**
Fixes #136 (I think)

### CODE REVIEW

File: duffy/database/model/tenant.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
 
-class Project(Base, CreatableMixin, RetirableMixin):
-    __tablename__ = "projects"
+class Tenant(Base, CreatableMixin, RetirableMixin):
+    __tablename__ = "tenants"
     __mapper_args__ = {"eager_defaults": True}
     id = Column(Integer, primary_key=True, nullable=False)
     name = Column(UnicodeText, unique=True, nullable=False)
+    is_admin = Column(Boolean, nullable=False)
```
Comment (nphilipp) [inline duffy/database/model/tenant.py]: Let's set a default:

```
    is_admin = Column(Boolean, nullable=False, default=False)
```

File: tests/database/test_model.py [hunk 2]
```diff
@@ -67,8 +68,10 @@ class TestSession(ModelTestBase):
     klass = model.Session
 
     def _db_obj_get_dependencies(self):
-        project = model.Project(name="My Other Project", ssh_key="my other public SSH key")
-        return {"project": project}
+        tenant = model.Tenant(
+            name="My Other Tenant", is_admin=False, ssh_key="my other public SSH key"
```
Comment (nphilipp) [inline tests/database/test_model.py]: With the default above, we wouldn't have to set `is_admin` explicitly here.

File: tests/database/test_model.py [hunk 3]
```diff
@@ -116,8 +119,8 @@ class TestSessionNode(ModelTestBase):
     attrs = {"distro_type": "CentOS", "distro_version": "8Stream"}
 
     def _db_obj_get_dependencies(self):
-        project = model.Project(name="World Domination", ssh_key="Muahahahaha!")
-        session = model.Session(project=project)
+        tenant = model.Tenant(name="World Domination", is_admin=True, ssh_key="Muahahahaha!")
```
Comment (nphilipp) [inline tests/database/test_model.py:120]: 👍😁

**Review by nphilipp — CHANGES_REQUESTED** [review]

File: tests/database/test_model.py (hunk 3 above)
Comment (gridhead) [reply tests/database/test_model.py:120]: I reckon they need that admin access, if they want to dominate the world and so is the column. ;P

**Comment by gridhead** [comment]
Should be good enough for another look, @nphilipp. Please let me know if more changes are needed. :)

**Review by nphilipp — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**